examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,430 @@
+# The model learnt is far from being a good model making accurate predictions:
+# the R squared score is very low.
+# As interpretation tools characterize model rather than the generative process
+# of the data itself, it needs to be emphasized that interpretations are
+# correct if the model is correct as well.
+# In this case, we are more interested in providing a methodology than in
+# having a good description of the data: a bad example could illustrate the
+# importance of cross checking the results.
+#
+# Interpreting coefficients
+# .........................
+#
+# First of all, we can plot the values of the coefficients of the regressor we
+# have fitted.
+
+feature_names = (model.named_steps['columntransformer']
+                      .named_transformers_['onehotencoder']
+                      .get_feature_names(input_features=categorical_columns))
+feature_names = np.concatenate(
+    [feature_names, binary_columns, numerical_columns])
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_,
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# Soon we realize that we cannot compare different coefficients since the
+# features have different natural scales and hence value ranges
+# because of their different unit of measure.
+# For instance, the AGE coefficient is expressed in $/hours/living years
+# while the EDUCATION is expressed in $/hours/years of education.
+# This is evident if we compare feature standard deviations.
+
+X_train_preprocessed = pd.DataFrame(
+    model.named_steps['columntransformer'].transform(X_train),
+    columns=feature_names
+)
+X_train_preprocessed.std(axis=0).plot(kind='barh', figsize=(9, 7))
+plt.title('Features std. dev.')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# We should then multiply the coefficients by the standard deviation and we
+# will be able to better compare them: in that way, we emphasize that the
+# greater the variance of a feature, the larger the impact of the corresponding
+# coefficent on the output, all else being equal.
+
+coefs = pd.DataFrame(
+    model.named_steps['transformedtargetregressor'].regressor_.coef_ *
+    X_train_preprocessed.std(axis=0),
+    columns=['Coefficients'], index=feature_names
+)
+coefs.plot(kind='barh', figsize=(9, 7))
+plt.axvline(x=0, color='.5')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# The plot above tells us that an increase of the AGE will induce a decrease
+# of the WAGE when all other features remain constant, for instance at a
+# constant EXPERIENCE. An increase of the EXPERIENCE will induce an increase
+# of the WAGE when all other features remain constant.
```

Comment:
This is something we could have deducted from the first plot already (and in a more interpretable way IMHO).

I think this last plot should lead to a different observation to be compelling. That would also help me understand what the goal is when we multiply by the std dev.

CC @glemaitre @GaelVaroquaux

Reply:
The goal when multiplying by the std dev is indeed not related to the effect of conditioning. Hence, maybe we should move the comment on conditional interpretations above.

The importance of the scaling is to be able to compare the importance of the different features. For instance, if I ask you, looking at the first plot: which are the most important features that impact wage, you would probably tell me: first union, and second sex (which would be pretty depressing). However, it would be misleading, as both union and sex vary on a small scale: they are either 0 or 1. On the other hand, a feature like age varies a lot more: several decades. Hence, across the dataset, the impact of age is much more visible on wage. For this reason, we multiply the coefficients by a number that captures the scale of variation of each feature, the standard deviation.

The paragraph that I wrote above could be added (with any modification needed) to the example. Before, or after the second plot.

Reply:
Also, I think that I'd like to see the word "conditional dependence", where we describe the phenomenon of interpreting the impact of age with constant experience, as it is the proper term in statistics, and can help people relate to things that they read or know (but the current non-technical wording should really stay).

Reply:
Should/could "marginal dependence" be used when describing the dependence of age without fixing the other variables?